### DISCUSSION TITLE
Re-Open WinUI3 app after a crash

### DISCUSSION THREAD

**Discussion by omar-awwad** [microsoft/microsoft-ui-xaml#9508] [Q&A]
We have a WinUI3 App that runs on a Kiosk device. 
Sometimes the app crashes, and we are working on ways to know the source of the crash.
Until then, is there any way to relaunch the app when crashes happen ?

**Comment by myokeeh** [comment]
Are you specifically using the Kiosk or Assigned access feature of Windows? If so, I believe it automatically restarts the app by default.

**Reply by omar-awwad** [reply]
I'm using normal winui 3 packaged app installed on windows 10 machine  ( not running kiosk mode)

**Comment by DarranRowe** [comment]
From what I can tell, [RegisterApplicationRestart](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-registerapplicationrestart) should work. The [MSIX documentation states](https://learn.microsoft.com/en-us/windows/msix/non-store-developer-updates#automatically-restarting-your-app-after-an-update) that a non UWP packaged application can use it to restart for updates. This is also used to restart after a crash.
I'll double check to be sure.

--Edit--
I take that back. It works as expected with unpackaged applications, but it doesn't automatically relaunch a packaged application.
